uniswap-factory-pool-monitoring/actions/example.ts

```diff
@@ -0,0 +1,44 @@
+import { Network as ActionsNetwork, Context, AlertEvent } from '@tenderly/actions';
+import { Network as SDKNetwork, Tenderly } from '@tenderly/sdk';
+const ethers = require('ethers');
+
+// Do not change function name.
+const actionFn = async (context: Context, alertEvent: AlertEvent) => {
+
+	const provider = new ethers.providers.JsonRpcProvider(context.gateways.getGateway(ActionsNetwork.MAINNET));
+	const key = await context.secrets.get('ACCESS-KEY')
+
+	const accountSlug = '';
+	const projectSlug = '';
+	const tagName = 'pool';
+
+	const tenderly = new Tenderly({
+		accountName: accountSlug,
+		projectName: projectSlug,
+		accessKey: key,
+		network: SDKNetwork.MAINNET, // Replace with the appropriate network
+	});
+
+	const resp = await provider.send("tenderly_traceTransaction", [`${alertEvent.hash}`]);
+	const poolCreatedLogs = resp.logs.filter((log: { name: string; }) => log.name === 'PoolCreated');
+	const poolInput = poolCreatedLogs[0].inputs.find((input: { name: string; }) => input.name === 'pool');
+	const poolAddress = poolInput.value;
+
+	async function addContract(childContractAddress: string) {
```

Comment:
There's no need for this to be a named function defined in this scope. Either:
- Just inline the code
- Define it outside of this scope